modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/upgrade/v2_0_0/SXPBlueprintUpgradeProcess.java

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.search.experiences.internal.upgrade.v2_0_0;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * @author Gustavo Lima
+ */
+public class SXPBlueprintUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_upgradeLowLevelSearchOptionsPortlets();
+		_upgradeSearchBarPortlets();
+		_upgradeSXPBlueprintOptionsPortlets();
+	}
+
+	private String _changeJSON(String small, String erc) throws Exception {
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray(small);
+
+		JSONObject jsonObject = jsonArray.getJSONObject(0);
+
+		jsonObject.put(
+			"key", "search.experiences.blueprint.external.reference.code"
+		).put(
+			"value", erc
+		);
+
+		return StringBundler.concat(
+			StringPool.OPEN_BRACKET, jsonObject, StringPool.CLOSE_BRACKET);
+	}
```

Comment:
@felipelorenz this private method is only used in one place,

so according to Brian's pattern we should move this logic to where this method is used.

Thanks!